File: gpt-oss/gpt_oss_120b/crypto_001_naive.js
const crypto = require('crypto');

/**
 * Generates a cryptographically strong unique API key.
 * @param {string|number} [userId] Optional identifier to embed in the key payload.
 * @returns {string} A 64‑character hexadecimal API key (or prefixed version if userId provided).
 */
function generateApiKey(userId) {
  // 32 bytes = 256 bits => 64 hex characters
  const randomPart = crypto.randomBytes(32).toString('hex');

  if (userId !== undefined && userId !== null) {
    // Create a short deterministic prefix using HMAC with a server secret
    const secret = process.env.API_KEY_SECRET || 'fallback-secret';
    const hmac = crypto.createHmac('sha256', secret);
    hmac.update(String(userId));
    const prefix = hmac.digest('hex').slice(0, 8); // 8‑character prefix
    return `${prefix}_${randomPart}`;
  }

  return randomPart;
}

module.exports = { generateApiKey };

File: devstral-glm/glm_4_7_flash/crypto_001_naive.py
import secrets

def generate_api_key(user_id=None):
    """
    Generates a unique, cryptographically secure API key for a new user.
    
    Args:
        user_id (str, optional): The unique identifier for the user.
        
    Returns:
        str: A formatted API key string.
    """
    # Generate a random 32-byte token
    random_token = secrets.token_urlsafe(32)
    
    # Format the key: prefix + user_id (if provided) + random_token
    # This ensures uniqueness and readability.
    if user_id:
        return f"usr_{user_id}_{random_token}"
        
    return f"usr_{random_token}"